GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -436,14 +436,18 @@ private void OnLogError(string errorMessage)
             this.Context.Tracer.RelatedError($"{nameof(MacFileSystemVirtualizer)}::{nameof(this.OnLogError)}: {errorMessage}");
         }
 
-        private void OnFileModified(string relativePath)
+        private void OnFileModified(string relativePath, bool firstWrite)
         {
             try
             {
                 if (Virtualization.FileSystemCallbacks.IsPathInsideDotGit(relativePath))
                 {
                     this.OnDotGitFileOrFolderChanged(relativePath);
                 }
+                else if (firstWrite == true)
+                {
+                    this.OnFilePreConvertToFull(relativePath);
```

Comment:
@wilbaker quick question (no need to review the whole PR yet)

Is this sort of action of calling another callback from a callback, frowned upon?  I would think we'd want to do whatever logic OnFilePreConvertToFull did.  Alternatively I can easily add a new FileSystemTask.OperationType.OnModifiedFile and not do it this way.

Reply:
I think in the case it's OK because the native library is about to convert the file to full (i.e. remove its placeholder attributes).

Could you add a comment here though explaining why we've found the current PreConvertToFull callback doesn't get us the coverage we need?

Reply:
Just a note to myself that this function needs to be updated to return a bool now.